genie-proto/src/main/proto/genie.proto

```diff
@@ -48,3 +48,86 @@ message AgentRegistrationResponse {
     bool agentAccepted = 3;
     string serverMessage = 4;
 }
+
+//
+// Job Specification APIs
+//
+service JobSpecificationService {
+    rpc resolveJobSpecification (ResolveJobSpecificationRequest) returns (JobSpecificationResponse);
+    rpc getJobSpecification (GetJobSpecificationRequest) returns (JobSpecificationResponse);
+}
+
+message Criterion {
+    repeated string tags = 1;
+    string id = 2;
+    string name = 3;
+    string status = 4;
+}
+
+message JobMetadata {
+    string id = 1;
+    string name = 2;
+    string user = 3;
+    string version = 4;
+    string description = 5;
+    repeated string tags = 6;
+    string metadata = 7;
+    string email = 8;
+    string grouping = 9;
+    string grouping_instance = 10;
+    string setup_file = 11;
+    repeated string configs = 12;
+    repeated string dependencies = 13;
+    repeated string command_args = 14;
+    bool is_interactive = 15;
+}
+
+message ExecutionResourceCriteria {
+    repeated Criterion cluster_criteria = 1;
+    Criterion command_criterion = 2;
+    repeated string application_ids = 3;
```

Comment:
Mh, is this right? Application IDs in the specification request?

Reply:
It's a feature present in Genie 3 requested by Dan's team that allows them to override defaults if need be. I do not expect most users to use this feature but it is there for power users and thus carried over here as well.